dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1115,29 +1135,31 @@ class IndexConsumer : public ASTConsumer,
       StringRef searchPath,
       StringRef relativePath,
       const Module *imported) {
-    PresumedLoc presumedHashLoc;
-    FileInfo *target, *source;
-    SourceLocation targetBegin, targetEnd;
+    PresumedLoc presumedHashLoc = sm.getPresumedLoc(hashLoc);
```

Comment:
Why move all this up into the declarations? Does it have a memory leak consequence? It was pushed into the ifs as a speed optimization.